tket/src/Circuit/basic_circ_manip.cpp

```diff
@@ -494,17 +530,15 @@ void Circuit::rewire(
     port_t port2 = get_target_port(preds[i]);
     Vertex old_v1 = source(preds[i]);
     Vertex old_v2 = target(preds[i]);
+
     if (insert_type == EdgeType::Boolean) {
-      if (replace_type != EdgeType::Classical) {
-        throw CircuitInvalidity(
-            "Cannot rewire; Boolean needs a classical value "
-            "to read from");
-      }
+      // Cannot rewire; Boolean needs a classical value to read from
+      TKET_ASSERT(replace_type == EdgeType::Classical);
+
       add_edge({old_v1, port1}, {new_vert, i}, insert_type);
     } else {
-      if (insert_type != replace_type) {
```

Comment:
This change was only made to reach the needed branch coverage:
The line was never reached in the tests and could only be reached by changing the code, I think it is the right thing to have an TKET_ASSERT in this case, replacing the throw exception.